src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java

```diff
@@ -368,6 +382,12 @@ public void onComplete() {
         return buildLogEntries;
     }
 
+    private void checkPath(String path) {
+        if (path == null || path.contains("..") || !path.matches("[a-zA-Z0-9_*./-]+")) {
```

Comment:
here you could make use of the method `stripIllegalCharacters` in `StringUtil` to improve it and reuse code a bit :D